tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3634Test.php

```diff
@@ -30,8 +27,17 @@
  */
 class DDC3634Test extends OrmFunctionalTestCase
 {
+    private LastInsertIdMocker $idMocker;
+
     protected function setUp(): void
     {
+        $this->idMocker = new LastInsertIdMocker();
+        $config         = new Configuration();
+        $config->setMiddlewares([new LastInsertIdMockMiddleware($this->idMocker)]);
```

Comment:
This still looks like too much code to mock a single value returned by `Connection::lastInsertId()`. Would it make sense to use PHPUnit mocks for that? Or the test uses other Connection APIs for which we want to use the default implementation?

Reply:
Yes, the test persists entities in a real database. What we want to test is if the ORM handles very large IDs returned as string correctly. If you have an idea to simplify this partial mock setup, I'm all ears.

Reply:
It is indeed a functional test. In this case, it shouldn't use mocks. It should reproduce a real case instead.

For instance, initialize the autoincrement field with `PHP_INT_MAX` and insert a new row.

Reply:
> It is indeed a functional test. In this case, it shouldn't use mocks. It should reproduce a real case instead. BTW, what is the corresponding code change that it covers?

That would be this seven years old PR: #1346.

> For instance, initialize the autoincrement field with `PHP_INT_MAX` and insert a new row.

Does the schema manager give me a nice way to do that?

> Alternatively, since it's not a real functional test, it could be converted into a unit one and cover only the corresponding component w/o mocking the whole world.

Yes, a unit test would not be a bad idea, however since we need to make sure that UoW and persisters talk to each other correctly, I would not want to remove the functional test either way.

Reply:
> Does the schema manager give me a nice way to do that?

I don't think so, since there's no portable API for that.

> Yes, a unit test would not be a bad idea, however since we need to make sure that UoW and persisters talk to each other correctly, I would not want to remove the functional test either way.

This test doesn't cover a significant part for the functionality. Specifically, the driver returning a bigint. As we learned recently, SQLite returns integers as integers. Mocking ~~this~~ anything in an integration test is a bit naive IMO.